tests/integration/conftest.py

```diff
@@ -18,6 +20,9 @@ def pytest_addoption(parser):
     parser.addoption(
         '--redis-executable', default='../redis/src/redis-server',
         help='Name of redis-server executable to use.')
+    parser.addoption(
```

Comment:
Do you see a good reason to pull this into the test? Not feeling so easy about this unnecessary coupling.

Also, generating certificates every time in a temporary directory is unnecessarily slow I think. After all, it's not certificate handling that we're testing here.

Reply:
we aren't generating it for each test, we generate it once per test suite run.  the only thing we "slow down" is if you don't run a tls test (i.e. running a single test), it still generates it.  can probably avoid that with a "skip tls generation" flag for those cases.

Reply:
My main concern is we're coupling things here in a way that has little value, IMHO. It would seem better that the test configuration could point to a dir / specific files to use and pass as arguments, and that the user will be required to run the script if TLS tests are to be executed.